From a60f8fd877dffee16d598e3ac27c0632b4db4cec Mon Sep 17 00:00:00 2001
From: Dominggoes Isakh <drjisakh@gmail.com>
Date: Sun, 29 May 2022 11:34:09 +0200
Subject: [PATCH 19/20] Revert "camera: Only link and use
 vendor.qti.hardware.camera.device if specified"

This reverts commit fa826bb4c3faeb57c855d98a0d435655eeeeea24.

Change-Id: Ieb4872f6927196d4e0c130245df9dc9970e8e81a
---
 services/camera/libcameraservice/Android.bp               | 2 +-
 .../libcameraservice/device1/CameraHardwareInterface.cpp  | 2 --
 .../libcameraservice/device1/CameraHardwareInterface.h    | 8 --------
 3 files changed, 1 insertion(+), 11 deletions(-)

diff --git a/services/camera/libcameraservice/Android.bp b/services/camera/libcameraservice/Android.bp
index 3b047c844a..a070b92597 100644
--- a/services/camera/libcameraservice/Android.bp
+++ b/services/camera/libcameraservice/Android.bp
@@ -40,7 +40,6 @@ cc_library_shared {
     defaults: [
         "needs_camera_boottime_defaults",
         "no_cameraserver_defaults",
-        "qti_camera_device_defaults",
     ],
 
     // Camera service source
@@ -156,6 +155,7 @@ cc_library_shared {
         "android.hardware.camera.provider@2.6",
         "android.hardware.camera.provider@2.7",
         "android.hardware.camera.device@1.0",
+        "vendor.qti.hardware.camera.device@1.0",
         "android.hardware.camera.device@3.2",
         "android.hardware.camera.device@3.3",
         "android.hardware.camera.device@3.4",
diff --git a/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp b/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp
index 61cada61a9..3a18bbf043 100644
--- a/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp
+++ b/services/camera/libcameraservice/device1/CameraHardwareInterface.cpp
@@ -136,7 +136,6 @@ hardware::Return<void> CameraHardwareInterface::dataCallback(
     return hardware::Void();
 }
 
-#ifdef QTI_CAMERA_DEVICE
 hardware::Return<void> CameraHardwareInterface::QDataCallback(
         DataCallbackMsg msgType, uint32_t data, uint32_t bufferIndex,
         const vendor::qti::hardware::camera::device::V1_0::QCameraFrameMetadata& metadata) {
@@ -155,7 +154,6 @@ hardware::Return<void> CameraHardwareInterface::QDataCallback(
     sDataCb((int32_t) msgType, mem, bufferIndex, &md, this);
     return hardware::Void();
 }
-#endif
 
 hardware::Return<void> CameraHardwareInterface::dataCallbackTimestamp(
         DataCallbackMsg msgType, uint32_t data,
diff --git a/services/camera/libcameraservice/device1/CameraHardwareInterface.h b/services/camera/libcameraservice/device1/CameraHardwareInterface.h
index 4a20b6a597..1d58e9ba16 100644
--- a/services/camera/libcameraservice/device1/CameraHardwareInterface.h
+++ b/services/camera/libcameraservice/device1/CameraHardwareInterface.h
@@ -29,9 +29,7 @@
 #include <hardware/camera.h>
 
 #include <common/CameraProviderManager.h>
-#ifdef QTI_CAMERA_DEVICE
 #include <vendor/qti/hardware/camera/device/1.0/IQCameraDeviceCallback.h>
-#endif
 
 namespace android {
 
@@ -88,11 +86,7 @@ typedef void (*data_callback_timestamp_batch)(
 
 class CameraHardwareInterface :
         public virtual RefBase,
-#ifdef QTI_CAMERA_DEVICE
         public virtual vendor::qti::hardware::camera::device::V1_0::IQCameraDeviceCallback,
-#else
-        public virtual hardware::camera::device::V1_0::ICameraDeviceCallback,
-#endif
         public virtual hardware::camera::device::V1_0::ICameraDevicePreviewCallback {
 
 public:
@@ -402,12 +396,10 @@ private:
             hardware::camera::device::V1_0::DataCallbackMsg msgType,
             const hardware::hidl_vec<
                     hardware::camera::device::V1_0::HandleTimestampMessage>&) override;
-#ifdef QTI_CAMERA_DEVICE
     hardware::Return<void> QDataCallback(
             hardware::camera::device::V1_0::DataCallbackMsg msgType,
             uint32_t data, uint32_t bufferIndex,
             const vendor::qti::hardware::camera::device::V1_0::QCameraFrameMetadata& metadata) override;
-#endif
 
     /**
      * Implementation of android::hardware::camera::device::V1_0::ICameraDevicePreviewCallback
-- 
2.34.1

